### ISSUE TITLE
Consider wrapped exception for going beyond max_matches

### ISSUE THREAD

**Issue by kanishka-azimi** [gregwebs/haskell-sphinx-client#17]
If one uses a limit and offset that goes beyond the allowed max_matches, you get a low level Data.Binary error. Would be nice if you could get the underlying exception instead. Not important though.

Current error message:  ` ... : Data.Binary.Get.runGet at position 4: not enough bytes` 

Underlying error message in mysql: 
````
mysql> select * from note_core where match('note') limit 1000,1 option max_matches=1000;
ERROR 1064 (42000): query 0 error: offset out of bounds (offset=1000, max_matches=1000)
`````

Feel free to close this issue if not a priority.

I haven't tried extensively to reproduce this separate from the larger program where this exception is occurring. I am just assuming that it is coming from this package.

**Comment by paul-rouse** [comment]
This looks like a bug in this package's error handling.

There are two ways errors can be reported in the protocol: a response can have status OK but contain a QueryError (or QueryWarning), or the response status itself can be ERROR (or RETRY).  For example, if you give a non-existent index name, you get a QueryError, and it is reported properly.  However, in your case, it is an ERROR response, and it is *not* being correctly decoded.

The problem is that the (two occurrences of the) definition of `errorMessage` drops 4 bytes from the response, but these bytes are actually needed by `getTxt` to read the length of the message.  (The idea of dropping the 4 bytes probably came from the reference implementation https://github.com/sphinxsearch/sphinx/blob/master/api/sphinxapi.php, but in that code the length is not being used again.)

I have pushed a fix, which solves the problem as I reproduced it.  @kanishka-azimi can you check that the latest github version works for you?  @gregwebs can you see any situation in which my fix might be wrong?

**Comment by kanishka-azimi** [comment]
Awesome. Yeah, I had actually drilled that deep the last time we got such an error and noticed something about not parsing the protocol right, but I didn't see the investigation out to a final answer like you did.

>  @kanishka-azimi can you check that the latest github version works for you?

Lazily, I will defer this to @bobgru or @danchoi . Can one of you check this sometime?

**Comment by paul-rouse** [comment]
@gregwebs could you please give me access to sphinx on hackage (I am "paulrouse" without a hyphen there).  Then I can make a minor-version release with this fix.  Thanks.

**Comment by gregwebs** [comment]
I just tried to add you on hackage, but after a successful form submit you aren't showing up. I will try again later.

**Comment by paul-rouse** [comment]
The wonderful ways of Hackage!  Bryan O'Sullivan had the same problem adding me to the mysql packages the other day.  I don't know whether it was just waiting which solved it, or the fact that he emailed Duncan Coutts.

**Comment by paul-rouse** [comment]
@gregwebs  OK, I'm there - thank you very much!

**Comment by paul-rouse** [comment]
Released in version 0.6.0.2
